packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -567,8 +569,16 @@ List<String> _validateList<T>(Object? yamlList, String context, String typeAlias
     }
   }
 
-  return errors;
+  return errors.isEmpty ? (List<T>.from(yamlList), <String>[]) : (null, errors);
```

Comment:
uber-nit: re-use empty errors rather than constructing a new list:

```dart
  return errors.isEmpty ? (List<T>.from(yamlList), errors) : (null, errors);
```